### ISSUE TITLE
Ability to customise dialog width of annotations in the pt editor

### ISSUE THREAD

**Issue by vicbergquist** [sanity-io/sanity#2621]
**Is your feature request related to a problem? Please describe.**
Sometimes the dialog to edit an annotation in the pte is too small, it would be great if it was customisable from the schema.

**Describe the solution you'd like**
Expose dialog size options in schema when defining annotations.

**Describe alternatives you've considered**
Increase the default dialog size and always have it be bigger. Not optimal.

**Comment by dmorda** [comment]
I second this request and would love the ability to specify the preferred with of the dialog. In my case, I have an annotation to specify an internal link. So I highlight the text and then there is a dropdown to other pages on the site, blogs, etc. However, when I try to pick something the dialog is so skinny that it's hard to know what I'm picking.

As a workaround, I added this to sanity.json.

```
{
      "implements": "part:@sanity/components/dialogs/popover-style",
      "path": "theme/PopoverDialog.css"
}
```

I then copy and pasted the contents of the PopoverDialog.css from the [source file](https://github.com/sanity-io/sanity/blob/next/packages/%40sanity/base/src/__legacy/%40sanity/components/dialogs/PopoverDialog.css) into my `theme/PopoverDialog.css` and changed the max-width as follows.

```
@nest &[data-size='small'] {
    max-width: 25rem;
}
```

This is a bit of a hack, but it does the trick until we have a better way of specifying the preferred width.

**Comment by geball** [comment]
Resolved in `v2.23.0` ([changelog](https://www.sanity.io/changelog#78a287e12e26) and [docs](https://www.sanity.io/docs/configuration#4badce07ff78))

**Comment by tylermcrobert** [comment]
Following up on this - I'm on `3.0.0` and am not able to get this working. For reference: 

```
{
  name: 'internalLink',
  type: 'object',
  title: 'Internal link',
  fields: [
    {
      name: 'reference',
      type: 'reference',
      to: [{type: 'infoSubpage'}, {type: 'info'}],
    },
  ],
  options: {
    modal: {
      type: 'popover',
      width: 'medium', // the default is "small"
    },
  },
}
```
